src/Elcodi/Component/Plugin/Services/PluginManager.php

```diff
@@ -0,0 +1,261 @@
+use Symfony\Component\Yaml\Yaml;
+
+use Elcodi\Component\Configuration\Exception\ConfigurationParameterNotFoundException;
+use Elcodi\Component\Configuration\Services\ConfigurationManager;
+use Elcodi\Component\Plugin\Entity\Plugin;
+use Elcodi\Component\Plugin\Interfaces\PluginInterface;
+
+/**
+ * Class PluginManager
+ */
+class PluginManager
+{
+    /**
+     * @var array
+     *
+     * Kernel
+     */
+    protected $kernel;
+
+    /**
+     * @var ConfigurationManager
+     *
+     * Configuration manager
+     */
+    protected $configurationManager;
+
+    /**
+     * Construct
+     *
+     * @param KernelInterface      $kernel               Kernel
+     * @param ConfigurationManager $configurationManager Configuration Manager
+     */
+    public function __construct(
+        KernelInterface $kernel,
+        ConfigurationManager $configurationManager = null
+    )
+    {
+        $this->kernel = $kernel;
+        $this->configurationManager = $configurationManager;
+    }
+
+    /**
+     * Load templates
+     *
+     * @return array Templates found
+     *
+     * @throws ConfigurationParameterNotFoundException Parameter not found
+     * @throws Exception                               ConfigurationBundle not installed
+     */
+    public function loadPlugins()
+    {
+        if (!($this->configurationManager instanceof ConfigurationManager)) {
+
+            throw new Exception('You need to install ConfigurationBundle');
+        }
+
+        $plugins = $this
+            ->configurationManager
+            ->get('store.plugins');
+
+        $plugins = new ArrayCollection($plugins);
+        $bundles = $this->kernel->getBundles();
+        $bundlesFound = [];
+
+        /**
+         * We add new Plugins found but we don't touch old configurations
+         *
+         * @var Bundle $bundle
+         */
+        foreach ($bundles as $bundle) {
+
+            if ($bundle instanceof PluginInterface) {
+
+                $bundleName = $bundle->getName();
+                $bundleNamespace = $bundle->getNamespace();
+                $bundlesFound[] = $bundleNamespace;
+
+                $specification = array_merge(
+                    [
+                        'bundle'              => $bundleName,
+                        'namespace'           => $bundleNamespace,
+                        'name'                => 'Unnamed',
+                        'description'         => '',
+                        'version'             => 'Any',
+                        'author'              => 'Anonymous',
+                        'year'                => 'NaN',
+                        'url'                 => '',
+                        'fa_icon'             => 'gear',
+                        'configuration_route' => null,
+                        'enabled'             => false,
+                        'configuration'       => []
+                    ],
+                    $this->getPluginSpecification($bundle->getPath())
```

Comment:
Maybe we should not allow the plugin to override some keys? Like `bundle` or `namespace`.